src/main/java/org/epac/trace/config/JwtAuthConverter.java

```diff
@@ -0,0 +1,62 @@
+package org.epac.trace.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtClaimNames;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter;
+import org.springframework.stereotype.Component;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class JwtAuthConverter implements Converter<Jwt, AbstractAuthenticationToken> {
+
+    private final JwtGrantedAuthoritiesConverter jwtGrantedAuthoritiesConverter = new JwtGrantedAuthoritiesConverter();
+
+    @Value("${jwt.auth.converter.principle-attribute}")
+    private String principleAttribute;
+
+    @Value("${jwt.auth.converter.resource-id}")
+    private String resourceId;
+
+    @Override
+    public AbstractAuthenticationToken convert(@NonNull Jwt jwt) {
+        Collection<GrantedAuthority> authorities = Stream.concat(
+                jwtGrantedAuthoritiesConverter.convert(jwt).stream(),
+                extractResourceRoles(jwt).stream()
+        ).collect(Collectors.toSet());
+
+        return new JwtAuthenticationToken(
+                jwt,
+                authorities,
+                getPrincipleClaimName(jwt)
+        );
+    }
+
+    private String getPrincipleClaimName(Jwt jwt) {
+        String claimName = principleAttribute != null ? principleAttribute : JwtClaimNames.SUB;
+        return jwt.getClaim(claimName);
```

Comment:
verifier que le claimName exist deja ne concentre pas sur le token keycloak si tu récupérer une autre jwt il faut le systeme detecte que le jwt pas compatible et sans bloquer le systeme
if (!jwt.hasClaim(claimName)) {
    throw new IllegalArgumentException("JWT does not contain the expected claim: " + claimName);
}